aspnetcore/signalr/diagnostics.md

```diff
@@ -0,0 +1,206 @@
+
+### Other logging providers
+
+You can also use other logging providers such as Serilog, Seq, NLog, or any other logging system that integrates with `Microsoft.Extensions.Logging`. If your logging system provides an `ILoggerProvider`, you can register it with `.AddProvider`:
+
+[!code-csharp[Configuring a custom logging provider in .NET client](diagnostics/net-client-custom-log.cs?highlight=6)]
+
+### Controlling verbosity
+
+If the debug logs are too verbose, you can set the default minimum level to `Information` and just increase the logging level for SignalR logs. This can be done in code, in much the same was as on the server:
+
+[!code-csharp[Controlling verbosity in .NET client](diagnostics/logging-config-client-code.cs?highlight=8-10)]
+
+## Network traces
+
+> [!NOTE]
+> A network trace contains the full contents of every message sent by your application. **Never** post raw network traces from production applications to public forums like GitHub.
+
+If you encounter an issue, a network trace can sometimes provide a lot of helpful information. This is particularly useful if you are going to file an issue on our issue tracker.
+
+### Using Fiddler (preferred option)
+
+This method works for all apps.
+
+Fiddler is a very powerful tool for collecting HTTP traces. Install it from [telerik.com/fiddler](https://www.telerik.com/fiddler), launch it, and then run your application and reproduce the issue. Fiddler is available for Windows, and there are beta versions for macOS and Linux.
+
+If you connect using HTTPS, there are some extra steps to ensure Fiddler can decrypt the HTTPS traffic. See the [Fiddler Documentation](https://docs.telerik.com/fiddler/Configure-Fiddler/Tasks/DecryptHTTPS) for more details.
+
+Once you've collected the trace, you can export the trace by choosing "File" > "Save" > "All Sessions..." from the menu bar.
+
+![Exporting all sessions from Fiddler](diagnostics/fiddler-export.png)
+
+### Using tcpdump (macOS and Linux only)
+
+This method works for all apps.
+
+You can collect raw TCP traces using tcpdump by running the following command from a terminal window. You may need to be `root` or prefix the command with `sudo` if you get a permissions error:
+
+```
+tcpdump -i [interface] -w trace.pcap
+```
+
+Replace `[interface]` with the network interface you wish to capture on. Usually this is something like `/dev/eth0` (for your standard Ethernet interface) or `/dev/lo0` (for localhost traffic). See the `tcpdump` man page on your host system for more information.
+
+### Using a browser
+
+This method only works for browser-based apps.
+
+You can also use a browser to capture some network traffic.
+
+> [!NOTE]
+> Most browsers don't include WebSocket and Server-Sent Event messages in exported network traces. If you are using those transports, using a tool like Fiddler or TcpDump (described below) is a better approach.
```

Comment:
Out of curiosity. Which browsers *do* include WebSocket and Server-Sent Event messages in exported network traces? I do mind leaving it out of the doc though, since it may change.

Reply:
No browser :). Rewrote these paragraphs from above feedback and rephrased that section.